web/static/js/components/questionnaires/QuestionnaireEditor.jsx

```diff
@@ -39,7 +39,45 @@ class QuestionnaireEditor extends Component {
   }
 
   questionnaireAddStep(stepType) {
-    this.props.actions.addStep(stepType)
+    this.setState({
+      ...this.state,
+      addingStep: true
+    }, () => {
+      // Add the step then automatically expand it
+      this.props.questionnaireActions.addStep(stepType)
+    })
+  }
+
+  deleteStep() {
+    const currentStepId = this.state.currentStep
+
+    this.setState({
+      currentStep: null,
+      addingStep: false
+    }, () => {
+      this.props.questionnaireActions.deleteStep(currentStepId)
+    })
+  }
+
+  internalState(currentStep, addingStep = false) {
+    return {
+      currentStep,
+      addingStep
+    }
+  }
+
+  componentWillReceiveProps(newProps) {
+    // This feels a bit hacky, but it let's us expand the step we just created.
```

Comment:
If the stat stores what step is expanded (like it was before) it's super simple to implement and there's no need for a hack. Why not implement it that way?